pyfar/io/_codec.py

```diff
@@ -247,9 +247,9 @@ def _decode_object_json_aided(name, type_hint, zipfile):
     ObjType = _str_to_type(type_hint[1:])
     try:
         return ObjType._decode(obj_dict)
-    except AttributeError:
+    except AttributeError as e:
         raise NotImplementedError(
-            f'You must implement `{type}._decode` first.')
+            f'You must implement `{type}._decode` first.') from e
```

Comment:
Why do we use `{type}` instead of something like `ObjType.__name__` here? Maybe someone can explain.

Reply:
done, type was even wrong, thanks for noticing.